### PR TITLE
New shutdown-ipc tests

### PR DESCRIPTION
**Pull request by newhoggy**
Run with:

```bash
cabal test cardano-testnet --test-options '-p Spec.Spec.Shutdown'
```

### CODE REVIEW

**Comment by newhoggy** [comment]
This has been happening on Mac in Nix, but runs okay in GH:

```
          54 ┃   -- Run cardano-node with pipe as stdin.  Use 0 file descriptor as shutdown-ipc
          55 ┃   (mStdin, _mStdout, _mStderr, pHandle, _releaseKey) <- H.createProcess =<<
             ┃   │ CWD: Just "/private/tmp/nix-build-cardano-testnet-test-cardano-testnet-tests-1.33.0-check.drv-0/chairman"
             ┃   │ Command line: /nix/store/y9aaqwb03xf0gj4w63rxg4zfwmj6cd8n-cardano-node-exe-cardano-node-1.33.0/bin/cardano-node run --config /nix/store/imsgqhdsddghhxkmiwzz323m9rhyj6xd-cardano-node-src/configuration/cardano/mainnet-config.json --topology /nix/store/imsgqhdsddghhxkmiwzz323m9rhyj6xd-cardano-node-src/configuration/cardano/mainnet-topology.json --database-path /private/tmp/nix-build-cardano-testnet-test-cardano-testnet-tests-1.33.0-check.drv-0/chairman/test-944b3314754ef365/db --socket-path /private/tmp/nix-build-cardano-testnet-test-cardano-testnet-tests-1.33.0-check.drv-0/chairman/test-944b3314754ef365/node-1-socket --host-addr 127.0.0.1 --port 51932 --shutdown-ipc 0
          56 ┃     ( H.procNode
          57 ┃       [ "run"
          58 ┃       , "--config", projectBase </> "configuration/cardano/mainnet-config.json"
          59 ┃       , "--topology", projectBase </> "configuration/cardano/mainnet-topology.json"
          60 ┃       , "--database-path", tempAbsPath </> "db"
          61 ┃       , "--socket-path", tempAbsPath </> "node-1-socket"
          62 ┃       , "--host-addr", "127.0.0.1"
          63 ┃       , "--port", show @Int port
          64 ┃       , "--shutdown-ipc", "0"
          65 ┃       ] <&>
          66 ┃       ( \cp -> cp
          67 ┃         { IO.std_in = IO.CreatePipe
          68 ┃         , IO.std_out = IO.UseHandle hNodeStdout
          69 ┃         , IO.std_err = IO.UseHandle hNodeStderr
          70 ┃         , IO.cwd = Just tempBaseAbsPath
          71 ┃         }
          72 ┃       )
          73 ┃     )
          74 ┃
          75 ┃   H.threadDelay $ 10 * 1000000
          76 ┃
          77 ┃   mExitCodeRunning <- H.evalIO $ IO.getProcessExitCode pHandle
          78 ┃
          79 ┃   when (isNothing mExitCodeRunning) $ do
          80 ┃     H.evalIO $ IO.hClose hNodeStdout
          81 ┃     H.evalIO $ IO.hClose hNodeStderr
          82 ┃     H.cat nodeStdoutFile
          83 ┃     H.cat nodeStderrFile
          84 ┃
          85 ┃   mExitCodeRunning === Nothing
             ┃   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
             ┃   │ ━━━ Failed (- lhs) (+ rhs) ━━━
             ┃   │ - Just (ExitFailure 1)
             ┃   │ + Nothing
```

File: cardano-testnet/test/Main.hs [hunk 1]
```diff
@@ -34,6 +35,7 @@ tests = do
       , H.testProperty "Spec.Plutus.SubmitApi.TxInLockingPlutus" Spec.Plutus.SubmitApi.TxInLockingPlutus.hprop_plutus
       , ignoreOnWindows "Spec.Plutus.Direct.ScriptContextEquality"  Spec.Plutus.Direct.ScriptContextEquality.hprop_plutus_script_context_equality
       , ignoreOnWindows "Spec.Plutus.Direct.ScriptContextEqualityMint" Spec.Plutus.Direct.ScriptContextEqualityMint.hprop_plutus_script_context_mint_equality
+      , ignoreOnWindows "Spec.Shutdown" Spec.Shutdown.hprop_shutdown
```
Comment (Jimbo4350) [inline cardano-testnet/test/Main.hs:38]: Just put a comment saying that currently there is a blocking call on Windows that prevents graceful shutdown (or something to that effect). And that we currently aren't testing the shutdown IPC flag.

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

File: cardano-testnet/test/Main.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-testnet/test/Main.hs:39]: Done!

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**